bothan-api/server-cli/src/commands/start.rs

```diff
@@ -0,0 +1,196 @@
+) -> anyhow::Result<()> {
+    let log_level = &app_config.log.level;
+    tracing_subscriber::fmt()
+        .with_env_filter(format!("bothan_core={log_level},bothan_api={log_level}"))
+        .init();
+
+    let store = init_store(&app_config, registry, reset).await?;
+    let ipfs_client = init_ipfs_client(&app_config).await?;
+    let crypto_server = init_crypto_server(&app_config, store, ipfs_client).await?;
+
+    Server::builder()
+        .add_service(QueryServer::new(crypto_server))
+        .serve(app_config.grpc.addr)
+        .await?;
+
+    Ok(())
+}
+
+async fn init_store(
+    config: &AppConfig,
+    registry: Registry<Valid>,
+    reset: bool,
+) -> anyhow::Result<SharedStore> {
+    if reset {
+        let _ = std::fs::remove_dir_all(&config.store.path);
+    }
+
+    if !config.store.path.is_dir() {
+        create_dir_all(&config.store.path).with_context(|| "Failed to create home directory")?;
+    }
+
+    let mut store = SharedStore::new(registry, &config.store.path)
+        .await
+        .with_context(|| "Failed to create store")?;
+    debug!("Store created successfully at \"{:?}\"", &config.store.path);
+
+    if !reset {
+        store
+            .restore()
+            .await
+            .with_context(|| "Failed to restore store state")?;
+    }
+
+    Ok(store)
+}
+
+async fn init_ipfs_client(config: &AppConfig) -> anyhow::Result<IpfsClient> {
+    let ipfs_builder = IpfsClientBuilder::new(config.ipfs.endpoint.clone());
+    let ipfs_client = match &config.ipfs.authentication {
+        IpfsAuthentication::Header { key, value } => {
+            let header_name = HeaderName::from_str(key)?;
+            let header_value = HeaderValue::from_str(value)?;
+            ipfs_builder
+                .with_header(header_name, header_value)
+                .build()?
+        }
+        IpfsAuthentication::None => ipfs_builder.build()?,
+    };
+
+    Ok(ipfs_client)
+}
+
+async fn init_crypto_server(
+    config: &AppConfig,
+    store: SharedStore,
+    ipfs_client: IpfsClient,
+) -> anyhow::Result<CryptoQueryServer> {
+    let manager_store = SharedStore::create_manager_store(&store);
+
+    let stale_threshold = config.manager.crypto.stale_threshold;
+    let version_req = VersionReq::from_str(&format!("<={}", VERSION))
+        .with_context(|| "Failed to parse bothan version requirement")?;
+    let mut manager =
+        CryptoAssetInfoManager::new(manager_store, ipfs_client, stale_threshold, version_req);
+
+    init_crypto_workers(&mut manager, &store, &config.manager.crypto.source).await?;
+
+    Ok(CryptoQueryServer::new(manager))
+}
+
+async fn init_crypto_workers(
+    manager: &mut CryptoAssetInfoManager<'static>,
+    store: &SharedStore,
+    source: &CryptoSourceConfigs,
+) -> anyhow::Result<()> {
+    type Binance = BinanceWorkerBuilder;
+    type CoinGecko = CoinGeckoWorkerBuilder;
+
+    if let Some(opts) = &source.binance {
+        add_worker::<Binance>(manager, store, "binance", opts).await?;
```

Comment:
should `"binance"` / `"coingecko"` be part of AssetWorkerBuilder trait? like a function that returns static str